doc/api.rst

```diff
@@ -165,6 +165,14 @@ functions in their ``formatter`` specializations.
 
 .. _udt:
 
+Enforcing Compile-time Format Checks
+------------------------------------
```

Comment:
I think this doesn't need to be a separate section. Instead add a paragraph to the Compile-time Format String Checks section above.